projects/dotcms-field-elements/src/components/dot-tags/components/dot-autocomplete/dot-autocomplete.tsx

```diff
@@ -0,0 +1,150 @@
+                    onBlur={(event: FocusEvent) => this.handleBlur(event)}
+                    onKeyDown={(event: KeyboardEvent) => this.handleKeyDown(event)}
+                />
+            </Fragment>
+        );
+    }
+
+    @Watch('data')
+    watchData(): void {
+        if (!this.autocomplete) {
+            this.initAutocomplete();
+        }
+    }
+
+    private handleKeyDown(event: KeyboardEvent): void {
+        const value = this.getInputElement()['value'];
+
+        if (value && this.keyEvent[event.key]) {
+            this.keyEvent[event.key](value);
+        }
+    }
+
+    private handleBlur(event: FocusEvent): void {
+        setTimeout(() => {
+            if (document.activeElement.parentElement !== this.getResultList()) {
+                this.clean();
+                this.lostFocus.emit(event);
+            }
+        }, 0);
+    }
+
+    private clean(): void {
+        this.getInputElement()['value'] = '';
+        this.cleanOptions();
+    }
+
+    private cleanOptions(): void {
+        const resultList = this.getResultList();
+
+        if (resultList) {
+            Array.from(resultList.querySelectorAll('li'))
+                .forEach(child => child.remove()
+            );
+        }
+    }
+
+    private emitSelection(selection: string): void {
+        this.clean();
+        this.selection.emit(selection);
+    }
+
+    private getInputElement(): HTMLElement {
+        return this.el.querySelector(`#${this.id}`);
+    }
+
+    private initAutocomplete(): void {
+        // tslint:disable-next-line: no-unused-expression
+        this.autocomplete = new autoComplete({
+            data: {
+                src: async () => this.getData()
+            },
+            sort: (a, b) => {
+                if (a.match < b.match) {
+                    return -1;
+                }
+                if (a.match > b.match) {
+                    return 1;
+                }
+                return 0;
+            },
+            placeHolder: this.placeholder,
+            selector: `#${this.id}`,
+            threshold: this.threshold,
+            searchEngine: 'strict',
+            highlight: true,
+            maxResults: this.maxResults,
+            debounce: this.debounce,
+            resultsList: {
+                container: () => {
+                    return this.getResultListId();
+                },
+                destination: this.getInputElement(),
+                position: 'afterend'
+            },
+            resultItem: (data) => {
+                return `${data.match}`;
+            },
+            onSelection: (feedback) => this.emitSelection(feedback.selection.value)
+        });
+    }
+
+    private getResultList(): HTMLElement {
+        return this.el.querySelector(`#${this.getResultListId()}`);
+    }
+
+    private getResultListId(): string {
+        return `${this.id}_results_list`;
+    }
+
+    private async getData(): Promise<string[]> {
```

Comment:
because this method call this one:

https://github.com/dotCMS/core-web/pull/1024/files/c261baafae99df914c2feec7a2eb74119914c1d6#diff-93b128627b34d1016edb7c20813a80cbR146

data is a prop and should be the methos that request data to the back end so it is async